trie/triedb/pathdb/database.go

```diff
@@ -186,17 +208,50 @@ func New(diskdb ethdb.Database, config *Config) *Database {
 			log.Crit("Failed to disable database", "err", err) // impossible to happen
 		}
 	}
-	log.Warn("Path-based state scheme is an experimental feature")
+	var contexts []interface{}
+	if verkle {
+		contexts = append(contexts, "verkle", true)
+	}
+	log.Warn("Path-based state scheme is an experimental feature", contexts...)
 	return db
 }
 
+// reader implements the Reader interface, providing the functionalities to
+// retrieve trie nodes by wrapping the internal state layer.
+type reader struct {
+	layer layer
+}
+
+// Node implements trie.Reader interface, retrieving the node with specified
+// node info. Note the hash comparison will be skipped ff the provided hash
+// is empty, tailored for verkle slim format.
+func (r *reader) Node(owner common.Hash, path []byte, hash common.Hash) ([]byte, error) {
+	blob, loc, err := r.layer.node(owner, path, 0)
+	if err != nil {
+		return nil, err
+	}
+	// Skip the hash comparison if the hash is not provided, it's useful in
+	// verkle's slim format which doesn't store the hash reference to reduce
+	// read/write amplification.
+	if hash == (common.Hash{}) {
+		return blob, nil
+	}
+	h := newHasher()
+	defer h.release()
+
+	if got := h.hash(blob); got != hash {
```

Comment:
Is it necessary to check for hash consistency? This comes with significant overhead.